src/seedu/addressbook/data/person/Address.java

```diff
@@ -45,7 +85,8 @@ public String toString() {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                        && this.value.equals(((Address) other).value)); // state
```

Comment:
don't use the 'value' variable. Instead use the new objects blocks, streets, etc.